rust/xaynet-core/src/mask/seed.rs

```diff
@@ -57,21 +57,16 @@ impl MaskSeed {
     }
 
     /// Derives a mask of given length from this seed wrt the masking configurations.
-    pub fn derive_mask(
-        &self,
-        len: usize,
-        config_many: MaskConfig,
-        config_one: MaskConfig,
-    ) -> MaskObject {
+    pub fn derive_mask(&self, len: usize, config: MaskConfigPair) -> MaskObject {
         let mut prng = ChaCha20Rng::from_seed(self.as_array());
 
-        let rand_int = generate_integer(&mut prng, &config_one.order());
-        let scalar_mask = MaskUnit::new_unchecked(config_one, rand_int);
+        let rand_int = generate_integer(&mut prng, &config.unit.order());
+        let scalar_mask = MaskUnit::new_unchecked(config.unit, rand_int);
 
-        let rand_ints = iter::repeat_with(|| generate_integer(&mut prng, &config_many.order()))
+        let rand_ints = iter::repeat_with(|| generate_integer(&mut prng, &config.vect.order()))
```

Comment:
one question that just came to my mind, not related to your change: is the compiler smart enough to only call `order()` once and reuse the resulting value for all subsequent calls of the closure? it probably can't know that because the result of `order()` might theoretically change in between the closure calls. hence we should probably rewrite this to 
```suggestion
        let order = config.vect.order();
        let rand_ints = iter::repeat_with(|| generate_integer(&mut prng, &order))
```

Reply:
I believe you are right about that, nice suggestion.